src/main/java/tc/oc/pgm/match/MatchManagerImpl.java

```diff
@@ -45,7 +46,7 @@
   private final MapLibrary library;
   private final MapLoader loader;
 
-  private final Map<String, Match> matchById = new HashMap<>();
+  private final Map<String, Match> matchById = new ConcurrentHashMap<>();
```

Comment:
I presume this is because the `PingListener` is getting this on an async thread?

Reply:
exactly